aws/terminator/application_services.py

```diff
@@ -367,3 +367,41 @@ def name(self):
 
     def terminate(self):
         self.client.delete_document(Name=self.name)
+
+
+class CloudFrontDistribution(Terminator):
+    @staticmethod
+    def create(credentials):
+        def paginate_distributions(client):
+            return client.get_paginator('list_distributions').paginate().build_full_result()['DistributionList']['Items']
+        return Terminator._create(credentials, CloudFrontDistribution, 'cloudfront', paginate_distributions)
+
+    @property
+    def created_time(self):
+        return self.instance['LastModifiedTime']
+
+    @property
+    def name(self):
+        return self.instance['DomainName']
+
+    def terminate(self):
+        self.client.delete_distribution(Id=self.instance['Id'])
```

Comment:
Distributions need to be disabled before they can be deleted.  The update event (setting Enabled=False) must then fully propagate across the CDN before the distro can be deleted.  https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/HowToDeleteDistribution.html

This logic will need to be built into the terminate method for each class.  ie, check if the distro is Enabled=True, disable it if it is and move on.  If the distro is already Enabled=False, check if the Status=Deployed.  When both Enabled=False and State=Deployed, try to delete.

Reply:
done!! Thanks